packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -153,34 +155,87 @@ export class WaferMap extends FoundationElement {
     /**
      * @internal
      */
-    public render(): void {
-        this.renderQueued = false;
-        this.initializeInternalModules();
-        this.renderer?.drawWafer();
+    public setHoverData(
+        hoverWidth: number,
+        hoverHeight: number,
+        hoverOpacity: HoverDieOpacity,
+        hoverTransform: string
+    ): void {
+        this.hoverWidth = hoverWidth;
+        this.hoverHeight = hoverHeight;
+        this.hoverOpacity = hoverOpacity;
+        this.hoverTransform = hoverTransform;
     }
 
-    private queueRender(): void {
-        if (!this.$fastController.isConnected) {
-            return;
-        }
-        if (!this.renderQueued) {
-            this.renderQueued = true;
-            DOM.queueUpdate(() => this.render());
+    /**
+     * @internal
+     */
+    public update(): void {
```

Comment:
Part of my question was me not having taken the time to understand how the update tracker works yet. I took a bit closer look and saw that each property update only triggers one `requires***Update` accessor to be `true`. If that remains the case then I think this implementation is fine.

But suppose that changing the `dies` property actually required both `DiesRenderInfoUpdate` and `ScalesUpdate` (just a made up example). If `update()` is only called once then it will only process one of those two changes.

Assuming that's accurate and is a use case we want to support, it would be easy to fix by changing this method to use chained `if`s instead of `else if`s. But that's just an idea: I could be misunderstanding the intended use case.

Reply:
sorry for the late reply, the use case here is a bit different because we have a somewhat linear update flow, we do not have independent updates. so the higher order updates, like `DiesRenderInfoUpdate` in your example will trigger a full length update flow which contains the `ScalesUpdate` flow. These are not separate updates, just different starting points in the same update flow. we do not have the option to run only certain segments of the update because it snowballs. we can only chose the altitude